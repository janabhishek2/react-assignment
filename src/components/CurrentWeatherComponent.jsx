import React, { useEffect } from "react";
import "public/assets/css/currentWeather.css";
import {
  cityActions,
  useCityDispatch,
  useCityState,
} from "../context/cityContext";
import { getCurrentDataApi } from "../urls";
import axios from "axios";

function CurrentWeatherComponent() {
  const {
    city: selectedCity,
    cityCurrData,
    isCityCurrLoading,
  } = useCityState();
  const dispatch = useCityDispatch();
  const { lat, lon, id: selectedCityId } = selectedCity;
  const {
    weather = [],
    main: { temp } = {},
    id: cityDataId = "",
  } = cityCurrData || {};
  const { id = "" } = weather[0] || [];

  useEffect(() => {
    const fetchCurrentWeatherData = async () => {
      try {
        dispatch(cityActions.setCityCurrLoading(true));
        const { data } = await axios.get(getCurrentDataApi(lat, lon));
        if (data) {
          dispatch(cityActions.setCityCurrData(data));
        }
      } catch (err) {
        dispatch(cityActions.setCityCurrData({}));
      } finally {
        dispatch(cityActions.setCityCurrLoading(false));
      }
    };
    Object.keys(selectedCity).length > 0 &&
      selectedCityId !== cityDataId &&
      fetchCurrentWeatherData();
  }, [selectedCity, cityCurrData]);

  return (
    !isCityCurrLoading && (
      <div className="currentWeather__main margin__top_sm">
        <div className="currentWeather__cityName">
          {(selectedCity.nm || "").toUpperCase()}
        </div>
        <div
          className={`currentWeather__svg margin__top_sm wi wi-icon-${id}`}
        ></div>
        <div className="currentWeather__temperature margin__top_sm">{temp}</div>
      </div>
    )
  );

  return null;
}

export default CurrentWeatherComponent;
