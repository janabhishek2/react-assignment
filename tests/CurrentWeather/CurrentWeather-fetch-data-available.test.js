import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import CurrentWeatherComponent from "../../src/components/CurrentWeatherComponent";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

jest.mock("app/context/cityContext.js", () => ({
  useCityState: () => ({
    city: {
      id: 3038789,
      nm: "Abbeville",
      lat: 50.099998,
      lon: 1.83333,
    },
    cityCurrData: {},
    cityForecastData: {},
    isCityCurrLoading: false,
    isCityForecastLoading: false,
  }),
  useCityDispatch: () => {},
}));

describe("CurrentWeather", () => {
  test("CurrentWeather - CurrentWeather-fetch-data available", () => {
    const wrapper = mount(<CurrentWeatherComponent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
