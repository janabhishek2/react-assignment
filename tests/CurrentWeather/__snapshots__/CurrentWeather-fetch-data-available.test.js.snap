// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather CurrentWeather - CurrentWeather-fetch-data available 1`] = `
<CurrentWeatherComponent>
  <div
    className="currentWeather__main margin__top_sm"
  >
    <div
      className="currentWeather__cityName"
    >
      ABBEVILLE
    </div>
    <div
      className="currentWeather__svg margin__top_sm wi wi-icon-"
    />
    <div
      className="currentWeather__temperature margin__top_sm"
    />
  </div>
</CurrentWeatherComponent>
`;
